pepper-apis/src/main/java/org/broadinstitute/ddp/event/dsmtask/impl/updateprofile/UpdateProfileProcessor.java

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.ddp.event.dsmtask.impl.updateprofile;
+
+import org.broadinstitute.ddp.event.dsmtask.api.DsmTaskData;
+import org.broadinstitute.ddp.event.dsmtask.api.DsmTaskProcessorAbstract;
+
+/**
+ * Processes Dsm Task of taskType='UPDATE_PROFILE'.
+ * It updates person's email, first, and last name.
+ *
+ * <p>Does the following steps:
+ * <ul>
+ *     <li>updates user's email (checking if this email not used by another user);</li>
+ *     <li>updates user's first and last name.</li>
+ * </ul>
+ * NOTE: if any data (passed from Dsm) is null then it not updated.
```

Comment:
This doesn't sound like the semantics we wanted? I thought we were going to allow `null` as value, and we only skip updating if the _key_ is not present in json. (Although I'm not sure what allow setting email to `null` will even mean...)

Reply:
I reimplemented according to the semantics: "we were going to allow null as value, and we only skip updating if the key is not present in json."